reactive-streams-operators/src/main/java/io/smallrye/mutiny/streams/stages/FlatMapCompletionStageFactory.java

```diff
@@ -41,15 +41,15 @@ private FlatMapCompletionStage(Function<I, CompletionStage<O>> mapper) {
             return source.onItem().transformToUni((I item) -> {
                 if (item == null) {
                     // Propagate an NPE to be compliant with the reactive stream spec.
-                    return Uni.createFrom().failure(new NullPointerException());
```

Comment:
I'm clearly missing a distinction, but aren't the two statements equivalent?

Reply:
No, let's write it without artifices:

```
Uni.createFrom().failure(new NullPointerException()); // create the NPE when this line is executed

Uni.createFrom().failure(() -> new NPE())); // create the NPE when there is a subscription. If no subscription... no npe :-D 
```

Reply:
So we're actually calling different `failure()` methods, passing the exception vs a callback, which impacts the creation.

Makes sense, thanks for the clarification!